src/NUnitTestExtractor/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -148,6 +152,50 @@ public static void GetTests(IList<string> dlls, Level level, TextWriter writer)
                                         case Level.Function:
                                             data = methodInfo.DeclaringType + "." + methodInfo.Name;
                                             break;
+
+                                        case Level.TestCase:
+
+                                            data = methodInfo.DeclaringType + "." + methodInfo.Name + "(";
```

Comment:
Something isn't working in the TestCase level.
When I run: `.\NUnitTestExtractor.exe -l testcase C:\git\nova-integration-tests\Blueprint-Tests\bin\Debug\AccessControlTests.dll > tests.txt`

For the AccessControlTests.SessionsTests.PutSessionWithSessionToken_Verify200OK test there should be 4 testcases, but I only see these 3:

```
"C:\git\nova-integration-tests\Blueprint-Tests\bin\Debug\AccessControlTests.dll" | AccessControlTests.SessionsTests.PutSessionWithSessionToken_Verify200OK()
"C:\git\nova-integration-tests\Blueprint-Tests\bin\Debug\AccessControlTests.dll" | AccessControlTests.SessionsTests.PutSessionWithSessionToken_Verify200OK(1234)
"C:\git\nova-integration-tests\Blueprint-Tests\bin\Debug\AccessControlTests.dll" | AccessControlTests.SessionsTests.PutSessionWithSessionToken_Verify200OK("do_some_action", 1234)
```

1. It's including the test function without any parameters, which is shouldn't include.
2. It's missing the null parameters.

What you should do is, first write a test to cover this scenario (you can add a fake test DLL with all these type of testcase scenarios) and verify that the test fails with the code as it is now.  Seeing it fail is important because it tells you your test is doing what it should do.  Then fix the extractor code until your test passes.  That's called a "red green refactor".  http://www.jamesshore.com/Blog/Red-Green-Refactor.html